src/scanspec/specs.py

```diff
@@ -600,7 +610,94 @@ def bounded(
+class Range(Spec[Axis]):
+    """Linearly spaced frames with start and stop as first and last midpoints.
+
+    This class is intended to handle linearly spaced frames defined with a
+    specific step size.
+
+    .. seealso::
+        `Linspace`: For linearly spaced frames defined with a number of frames.
+
+    .. example_spec::
+
+        from scanspec.specs import Fly, Range
+
+        spec = Fly(Range("x", 1, 2, 0.25))
+    """
+
+    axis: Axis = Field(description="An identifier for what to move")
+    start: float = Field(description="Midpoint of the first point of the line")
+    stop: float = Field(description="Midpoint of the last point of the line")
+    step: float = Field(description="Step size")
+
+    def __post_init__(self):
+        if self.step == 0:
+            raise ValueError("step must be nonzero")
+
+    def axes(self) -> list[Axis]:  # noqa: D102
+        return [self.axis]
+
+    def _line_from_indexes(
+        self, indexes: npt.NDArray[np.float64]
+    ) -> dict[Axis, npt.NDArray[np.float64]]:
+        step = abs(self.step) * np.sign(self.stop - self.start)
+        first = self.start - step / 2
+        return {self.axis: indexes * step + first}
+
+    def calculate(  # noqa: D102
+        self, bounds: bool = False, nested: bool = False
+    ) -> list[Dimension[Axis]]:
+        step = abs(self.step)
+        distance = abs(self.stop - self.start)
+        # +1 to include start
+        num = int(distance // step) + 1
+        if np.isclose(step * num, distance):
+            # +1 to include stop
+            num = num + 1
+        return _dimensions_from_indexes(
+            self._line_from_indexes, self.axes(), num, bounds
+        )
+
+    @classmethod
+    def bounded(
+        cls: type[Range[Any]],
+        axis: OtherAxis = Field(description="An identifier for what to move"),
+        lower: float = Field(description="Lower bound of the first point of the line"),
+        upper: float = Field(description="Upper bound of the last point of the line"),
+        step: float = Field(description="Step size"),
+    ) -> Range[OtherAxis]:
+        """Specify a Range by extreme bounds instead of midpoints.
+
+        .. example_spec::
+
+            from scanspec.specs import Fly, Range
+
+            spec = Fly(Range.bounded("x", 1, 5, 2))
+        """
+        distance = abs(upper - lower)
+        direction = np.sign(upper - lower)
+        step = min(distance, abs(step))  # produce at least one frame
+        half_step = step / 2 * direction
+        start = lower + half_step
+        stop = upper - half_step
+        if stop == start:
+            # edge case with computed start and stop
+            stop = np.nextafter(start, np.inf * direction)
+        return cls(axis, start, stop, step)
+
+
+"""
+Defers wrapping function with validate_call until class is fully instantiated
+"""
+Range.bounded = validate_call(Range.bounded)  # type:ignore
+
+# Define alias for Range
+Line = Range
```

Comment:
This is for back compat, so shouldn't this be `Linspace` rather than `Range`?

```suggestion
Line = Linspace
```

Reply:
Yes, I mixed up `np.linspace` and `range` when drawing on the whiteboard...